File: Python/problem50.py
#Consecutive prime sum

def isPrime(n):
    limit = int(abs(n)**(1/2)+1)
    if(n == 1):
        return False
    for i in range(2,limit):
        if n % i == 0:
            return False
    else:
        return True

primes = []
i = 0
while len(primes) < 1000000 :
    if( isPrime(i) ):
        primes.append(i)
    i += 1

print("done gen. primes to", len(primes))

for i in range(0,len(primes)):

    Sum = 0
    index = 0

    while(not Sum >= primes[i]):
        Sum += primes[index]
        index += 1
        if(Sum == primes[i]):
            print(Sum,index)
            break

File: Java/problem50.java
import java.util.ArrayList;
public class problem50 {

	public static void main(String[] args) {
		ArrayList<Integer> primes = new ArrayList<Integer>();
		int i = 2;
		int LIMIT = 1000000;
		while(i < LIMIT) {
			if(isPrime(i)){
				primes.add(i);
			}
			i++;
		}
		System.out.println("Primes listed: " + primes.size());
		
		int currentSum = 0;
		int currentNumber = 6;
		
		for(int i1 = 0; i1<primes.size(); i1++) {
			while(currentSum + currentNumber < LIMIT) {
				if(isPrime(currentNumber)) {
					currentSum  += currentNumber;
				}
				currentNumber++;
			}
		}
		
		System.out.println(currentSum);
		
	
		
		
	}
		
	
	public static boolean isPrime(int n) {
		if (n > 2 && n%2 == 0) {
	        return false;
	    }
	    int top = (int)Math.sqrt(n) + 3;
	    for(int i = 3; i < top; i+=2){
	        if(n % i == 0){
	            return false;
	        }
	    }
	    return true; 
	}
	
	public static int sumOfNumbersInList(int from, int to, ArrayList<Integer> list) {
		int sum = 0;
		
		
		for(int i = from; i<to; i++) {
			if(sum>1000000) {
				break;
			}
			sum += list.get(i);
		}
		return sum;
	}
}
